### PR TITLE
Fixes #29212 - support el8

### PR DESCRIPTION
**Pull request by wbclark**
This is a WIP which incorporates the PRs #826 & #827 .

### CODE REVIEW

**Comment by wbclark** [comment]
When I test locally with EL7 it's failing due to the known issue: https://projects.theforeman.org/issues/29429

With EL8, an error occurs  when running puppet apply: `Error: Could not find a suitable provider for foreman_config_entry`

**Comment by mmoll** [comment]
@wbclark please rebase :)

File: spec/acceptance/foreman_cli_plugins_spec.rb [hunk 1]
```diff
@@ -11,9 +11,9 @@ class { 'foreman::cli':
       password    => 'changeme',
     }
 
-    if $facts['osfamily'] == 'RedHat' {
-      include foreman::cli::ansible
-      include foreman::cli::azure
+    if $facts['os']['family'] == 'RedHat' {
+      include ::foreman::cli::ansible
+      include ::foreman::cli::azure
```
Comment (ekohl) [inline spec/acceptance/foreman_cli_plugins_spec.rb:16]: This is the old style we got rid of
```suggestion
      include foreman::cli::ansible
      include foreman::cli::azure
```

**Comment by ekohl** [comment]
In this module, you need to ensure the `en` locales are present due to how we configure postgresql. I wonder if we should ensure this in `foreman::install`.

**Comment by wbclark** [comment]
> In this module, you need to ensure the `en` locales are present due to how we configure postgresql. I wonder if we should ensure this in `foreman::install`.

I've opened #837 to do this for the RedHat family of operating systems when the database is managed.

**Comment by wbclark** [comment]
test failure is due to: https://projects.theforeman.org/issues/29429

**Comment by mmoll** [comment]
merged, thanks @wbclark!

**Pull request merged**